src/errors.h

```diff
@@ -49,6 +49,7 @@
 #define ERRCODE_TS_DATA_NODE_ALREADY_ATTACHED MAKE_SQLSTATE('T', 'S', '1', '7', '2')
 #define ERRCODE_TS_DATA_NODE_NOT_ATTACHED MAKE_SQLSTATE('T', 'S', '1', '7', '3')
 #define ERRCODE_TS_DATA_NODE_INVALID_CONFIG MAKE_SQLSTATE('T', 'S', '1', '7', '4')
+#define ERRCODE_TS_DATA_NODE_SELF_REFERENCE MAKE_SQLSTATE('T', 'S', '1', '7', '5')
```

Comment:
Do we need this new error code? Seems like a one-off corner-case, so maybe a more generic error suffices?

Reply:
Using the `ERRCODE_TS_DATA_NODE_INVALID_CONFIG` which seems to be used for similar situations at other places.